src/frontend/src/scheduler/hummock_snapshot_manager.rs

```diff
@@ -57,6 +57,13 @@ impl PinnedHummockSnapshot {
             },
         }
     }
+
+    pub fn only_checkpoint_visible(&self) -> bool {
```

Comment:
The return value of this func will be used as 2nd arg of `new(manager: WorkerNodeManagerRef, enable_barrier_read: bool)`.  Seems a little hard to get the context. May change `only_checkpoint_visible` to `read_committed` ?